.NET/Microsoft.Recognizers.Text.Number/Parsers/BaseIndianNumberParser.cs

```diff
@@ -0,0 +1,838 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Recognizers.Text.Number
+{
+    public class BaseIndianNumberParser : BaseNumberParser
+    {
+        private readonly bool isMultiDecimalSeparatorCulture = false;
+
+        private readonly bool isCompoundNumberLanguage = false;
+
+        public BaseIndianNumberParser(INumberParserConfiguration config)
+            : base(config)
+        {
+            this.Config = config as IIndianNumberParserConfiguration;
+
+            this.isMultiDecimalSeparatorCulture = config.IsMultiDecimalSeparatorCulture;
+            this.isCompoundNumberLanguage = config.IsCompoundNumberLanguage;
+
+            TextNumberRegex = BuildTextNumberRegex();
+        }
+
+        protected new IIndianNumberParserConfiguration Config { get; private set; }
+
+        protected new Regex TextNumberRegex { get; }
+
+        public override ParseResult Parse(ExtractResult extResult)
```

Comment:
Is the flow of the whole Parse substantially different from the BaseParser? Otherwise we should insert specific overridable points.

Reply:
Is there any special behaviour in this Parse?
Why not just override DigitNumberParse, FracLikeNumberParse, TextNumberParse, GetIntValue, and GetDigitalValue? Are there other tweeks besides replacing these methods that I didn't notice?